### PR TITLE
Feature/cxspa 3042 Saving billing address

### PR DESCRIPTION
**Pull request by rmch91**

### CODE REVIEW

File: feature-libs/checkout/base/core/facade/checkout-payment.service.ts [hunk 1]
```diff
@@ -88,20 +88,19 @@ export class CheckoutPaymentService implements CheckoutPaymentFacade {
             if (!paymentDetailsId) {
               throw new Error('Checkout conditions not met');
             }
-
             return this.checkoutPaymentConnector
               .setPaymentDetails(userId, cartId, paymentDetailsId)
               .pipe(
-                tap(() =>
+                tap(() => {
```
Comment (Matejk00) [inline feature-libs/checkout/base/core/facade/checkout-payment.service.ts]: Do we need such changes in this file?

File: feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts [hunk 2]
```diff
@@ -55,6 +55,14 @@ export interface CheckoutOccEndpoints {
...
    */
   deliveryModes?: string | OccEndpoint;
+  /**
+   * Endpoint for set billing address to cart
+   */
+  setBillingAddress?: string | OccEndpoint;
+  /**
+   * Endpoint for getting specific cart
+   */
+  getCart?: string | OccEndpoint;
```
Comment (Matejk00) [inline feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts]: I think we have already this endpoint in `feature-libs/cart/base/occ/model/occ-cart-endpoints.model.ts` called `cart`. Doesn't they do the same?

File: feature-libs/checkout/base/root/facade/checkout-billing-address.facade.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      methods: ['setBillingAddress'],
+      // TODO:#deprecation-checkout - remove once we remove ngrx
+      async: true,
+    }),
+})
+export abstract class CheckoutBillingAddressFacade {
+  /**
+   * Sets the billing address to the cart
+   */
+  abstract setBillingAddress(address: Address): Observable<unknown>;
```
Comment (Matejk00) [inline feature-libs/checkout/base/root/facade/checkout-billing-address.facade.ts:27]: Can we create a method which indicate the billing address state? To be consistent with other declarations. For this let's use `QueryState` interface.

File: integration-libs/opf/pipes/get-address-card-content.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  line1: 'Noname street',
+  line2: '',
+  town: 'Warsaw',
+  postalCode: '02651',
+  phone: '',
+  cellphone: '',
+  defaultAddress: false,
+};
+
+fdescribe('GetAddressCardContent', () => {
```
Comment (Matejk00) [inline integration-libs/opf/pipes/get-address-card-content.pipe.spec.ts]: fdescribe

File: integration-libs/opf/pipes/get-address-card-content.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (Matejk00) [inline integration-libs/opf/pipes/get-address-card-content.pipe.ts]: We don't need to create separate folder and separate module for pipes. Let's move this pipe to the component folder where this pipe is used and also put it to the declaration of such component module.

File: integration-libs/opf/components/opf-checkout-billing-address-form/opf-checkout-billing-address-form.component.ts [hunk 6]
```diff
@@ -4,141 +4,189 @@
...
       }
     });
   }
 
   editCustomBillingAddress() {
-    this.editBillingAddress$.next(true);
+    this.editBillingAddressSub.next(true);
+  }
+
+  protected getCountries(): void {
```
Comment (Matejk00) [inline integration-libs/opf/components/opf-checkout-billing-address-form/opf-checkout-billing-address-form.component.ts]: There is a lot of custom logic in this component. Do you mind to move some methods to the separate service, and in component just call those methods from such service?

File: feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.spec.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(cartData);
+          done();
+        });
+
+      const mockReq = httpMock.expectOne((req) => {
+        return (
+          req.method === 'PUT' &&
+          req.url === `orgUsers/${userId}/carts/${cartId}/addresses/billing`
```
Comment (Matejk00) [inline feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.spec.ts]: orgUsers? Iam not sure why we compare to B2B endpoint here.

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts (hunk 2 above)
Comment (rmch91) [reply feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts]: i don't see the endpoint for getting cart by identifier in cart, i think it is not used anywhere, that's why i implemented this in checkout. To be honest, i don't know how it should be implemented from architecture point here

File: feature-libs/checkout/base/core/facade/checkout-payment.service.ts (hunk 1 above)
Comment (rmch91) [reply feature-libs/checkout/base/core/facade/checkout-payment.service.ts]: changed

File: integration-libs/opf/pipes/get-address-card-content.pipe.spec.ts (hunk 4 above)
Comment (rmch91) [reply integration-libs/opf/pipes/get-address-card-content.pipe.spec.ts]: removed

File: integration-libs/opf/pipes/get-address-card-content.pipe.ts (hunk 5 above)
Comment (rmch91) [reply integration-libs/opf/pipes/get-address-card-content.pipe.ts]: removed

File: integration-libs/opf/components/opf-checkout-billing-address-form/opf-checkout-billing-address-form.component.ts (hunk 6 above)
Comment (rmch91) [reply integration-libs/opf/components/opf-checkout-billing-address-form/opf-checkout-billing-address-form.component.ts]: moved to service

File: feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.spec.ts (hunk 7 above)
Comment (rmch91) [reply feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.spec.ts]: changed

File: feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts (hunk 2 above)
Comment (rmch91) [reply feature-libs/checkout/base/occ/model/occ-checkout-endpoints.model.ts]: those endpoints were removed and i am using active cart now

File: feature-libs/checkout/base/root/facade/checkout-billing-address.facade.ts (hunk 3 above)
Comment (rmch91) [reply feature-libs/checkout/base/root/facade/checkout-billing-address.facade.ts:27]: left as it is for now

File: feature-libs/cart/base/root/models/cart.model.ts [hunk 8]
```diff
@@ -79,6 +79,7 @@ export interface Cart {
   orderDiscounts?: Price;
   paymentInfo?: PaymentDetails;
   paymentType?: PaymentType;
+  paymentAddress?: Address;
```
Comment (Matejk00) [inline feature-libs/cart/base/root/models/cart.model.ts:82]: Shouldn't we use everywhere 'billingAddress' term?

File: feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+export class OccCheckoutBillingAddressAdapter
+  implements CheckoutBillingAddressAdapter
+{
+  constructor(
+    protected http: HttpClient,
+    protected occEndpoints: OccEndpointsService,
+    protected converter: ConverterService
+  ) {}
+
+  public setAddress(
```
Comment (Matejk00) [inline feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.ts]: What about rename to 'setBillingAddress'?

File: feature-libs/checkout/base/core/connectors/checkout-billing-address/checkout-billing-address.adapter.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+export abstract class CheckoutBillingAddressAdapter {
+  /**
+   * Abstract method used to set address for billing
+   *
+   * @param userId
+   * @param cartId
+   * @param addressId
+   */
+  abstract setAddress(
```
Comment (Matejk00) [inline feature-libs/checkout/base/core/connectors/checkout-billing-address/checkout-billing-address.adapter.ts]: What about change name to 'setBillingAddress'?

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: feature-libs/checkout/base/core/connectors/checkout-billing-address/checkout-billing-address.adapter.ts (hunk 10 above)
Comment (rmch91) [reply feature-libs/checkout/base/core/connectors/checkout-billing-address/checkout-billing-address.adapter.ts]: changed

File: feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.ts (hunk 9 above)
Comment (rmch91) [reply feature-libs/checkout/base/occ/adapters/occ-checkout-billing-address.adapter.ts]: changed

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**